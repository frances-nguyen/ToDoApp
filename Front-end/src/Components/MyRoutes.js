import React from 'react'
import { Route, Switch } from 'react-router-dom'
import LoginPage from './Global/LoginPage'
import WelcomePage from './Welcome/WelcomePage'
import ErrorPage from './ErrorPage'
import TodoList from './ToDoList/TodoList'
import LogOut from './Global/LogOut'
import AuthenticatedRoute from './AuthenticationService/AuthenticatedRoute'
import Todo from './ToDoList/Todo'

const MyRoutes = () => (
  <Switch>
    <Route exact path="/" component={LoginPage} />
    <Route path="/login" component={LoginPage} />
    <AuthenticatedRoute path="/welcome/:name" component={WelcomePage} />
    <AuthenticatedRoute path="/todolist" component={TodoList} />
    <AuthenticatedRoute path="/logout" component={LogOut} />
    <AuthenticatedRoute path="/todos/:id" component={Todo} />
    <Route component={ErrorPage} />
  </Switch>
)

export default MyRoutes