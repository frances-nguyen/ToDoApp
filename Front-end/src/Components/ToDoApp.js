import React, { Component } from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
//import Authentication from './AuthenticationService/Authentication'
import MyRoutes from './MyRoutes'
import Header from './Global/Header'
import Footer from './Global/Footer'

class ToDoApp extends Component {
  render() {
    return (
      <div>
        <Router>
          <Header />
          <MyRoutes />
          <Footer />
        </Router>
      </div>
    )
  }
}

export default ToDoApp
