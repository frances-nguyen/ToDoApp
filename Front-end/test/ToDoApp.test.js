import React from 'react'
import ToDoApp from '../src/Components/ToDoApp'
import MyRoutes from '../src/Components/MyRoutes'
import renderer from 'react-test-renderer'

describe('<ToDoApp /> rendering', () => {
  it('renders correctly', () => {
    let tree = renderer.create(<ToDoApp />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders <MyRoutes />', () => {
    let wrapper = shallow(<ToDoApp />)
    expect(wrapper.find(MyRoutes)).toHaveLength(1)
  })
})
