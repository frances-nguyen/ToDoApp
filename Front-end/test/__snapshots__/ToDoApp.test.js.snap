// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoApp /> rendering renders correctly 1`] = `
<div>
  <header>
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <div>
        <a
          className="navbar-brand"
          href="/"
          style={
            Object {
              "color": "#e3f2fd",
            }
          }
        >
          To Do
        </a>
      </div>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarNav"
        data-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          className="navbar-nav"
        >
          <li>
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <a
            className="nav-link"
            href="/todolist"
            onClick={[Function]}
          >
            To Do
          </a>
        </ul>
        <ul
          className="navbar-nav navbar-collapse justify-content-end"
        >
          <a
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Log In
          </a>
          <a
            className="nav-link"
            href="/logout"
            onClick={[Function]}
          >
            Log Out
          </a>
        </ul>
      </div>
    </nav>
  </header>
  <div>
    Username:
     
    <input
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    Password:
     
    <input
      name="password"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Login
    </button>
  </div>
  <div>
    <hr />
    Footer
  </div>
</div>
`;
